api/services/CasebookService.js

```diff
@@ -1,52 +1,93 @@
+const queryParamObjToStr = require('../helpers/queryParamObjToStr');
+const config = require('../../config/environment-variables');
+
+const {
+    hmacKey,
+    casebookCertificate: cert,
+    casebookKey: key,
+    customURLs,
+} = config;
 
 const CasebookService = {
-    get(options) {
-        return CasebookService._returnRequestMethods('get', options);
-    },
+    getApplicationStatus(applicationReference) {
+        const queryParamsObj = {
+            timestamp: Date.now().toString(),
+            applicationReference,
+        };
+        const casebookRequestBase =
+            CasebookService._createBaseRequest(queryParamsObj);
 
-    post(options) {
-        return CasebookService._returnRequestMethods('post', options);
+        return casebookRequestBase.get(customURLs.applicationStatusAPIURL, {
+            params: queryParamsObj,
+        });
     },
 
-    _returnRequestMethods(method, options) {
-        const requestType = options.promise ? requestPromise : request;
-        delete options.promise;
-
-        const authParams = CasebookService._createAuthParams(options.qs);
-        const optionsWithAuthParams = { ...options, ...authParams };
+    getApostilleDownload(apostilleReference) {
+        const queryParamsObj = {
+            timestamp: Date.now().toString(),
+            apostilleReference,
+        };
+        const casebookRequestBase =
+            CasebookService._createBaseRequest(queryParamsObj);
 
-        return requestType[method](optionsWithAuthParams);
+        return casebookRequestBase.get(customURLs.apostilleDownloadAPIURL, {
+            params: queryParamsObj,
+            responseType: 'stream',
+        });
     },
 
-    _createAuthParams(queryParamsObj) {
-        const {
-            hmacKey,
-            casebookCertificate: cert,
-            casebookKey: key,
-        } = sails.config;
+    _createBaseRequest(paramsObj) {
+        const casebookRequestBase = axios.create({
+            baseURL: customURLs.casebookBaseUrl,
+            headers: {
+                'Content-Type': 'application/json; charset=utf-8',
+                'api-version': '4',
+                hash: CasebookService._addHashToHeader(paramsObj),
+            },
+            httpsAgent: new https.Agent({
+                cert,
+                key,
+                keepAlive: true,
+            }),
+            paramsSerializer: queryParamObjToStr,
+            transformRequest: [CasebookService._addHmacToQueryParam],
+        });
+
+        casebookRequestBase.interceptors.response.use(
+            (response) => {
+                if (response.status !== 200) {
+                    throw new Error(`Casebook returned ${response.statusCode}`);
+                }
+                return response;
+            },
+            (error) => {
+                return Promise.reject(error);
+            }
+        );
 
-        const queryStr = apiQueryString.stringify(queryParamsObj);
+        return casebookRequestBase;
+    },
```

Comment:
it was ok as it was 😄!! you shouldn't have to create a new instance each time you want to do a call.

```.js

const baseRequest = axios.create(..) 
baseRequest.interceptors.response.use(...)

const CasebookService = { 
  getApplicationStatus(applicationReference) {
    const query = {
      timestamp: Date.now().toString(),
      applicationReference,
    };
    const params = thatFunctionThatParsesIt(query)
    //.. 
    return casebookRequestBase.get(customURLs.applicationStatusAPIURL, {
             params,
             // if there are other axios options you want to override from the baseRequest, you can do so here
    });
  }
}
```

Reply:
(sorry!! I realise this might actually be a bit of a trial by fire since the request lib and fetch don't have this concept)

Reply:
No worries, it's all part of the learning experience 😁